java/vector/src/main/java/org/apache/arrow/vector/BitVectorHelper.java

```diff
@@ -179,6 +179,109 @@ public static int getNullCount(final ArrowBuf validityBuffer, final int valueCou
     return 8 * sizeInBytes - count;
   }
 
+  /**
+   * Tests if all bits in a validity buffer are 0.
+   * @param validityBuffer the validity buffer.
+   * @param valueCount the bit count.
+   * @return true if all bits are 0, and false otherwise.
+   */
+  public static boolean allBitsNull(final ArrowBuf validityBuffer, final int valueCount) {
+    if (valueCount == 0) {
+      return true;
+    }
+    final int sizeInBytes = getValidityBufferSize(valueCount);
+    // If value count is not a multiple of 8, then calculate number of used bits in the last byte
+    final int remainder = valueCount % 8;
+    final int fullBytesCount = remainder == 0 ? sizeInBytes : sizeInBytes - 1;
+
+    int index = 0;
+    while (index + 8 <= fullBytesCount) {
+      long longValue = validityBuffer.getLong(index);
```

Comment:
I've seen that these do bound checks on every call and too many of these can add up.

In https://github.com/apache/arrow/pull/4847, I did one boundary check at the beginning of the function and bypassed the rest.

Reply:
@pravindra thanks for the suggestion.

I agree that it is a good idea to do the boundary check once at the beginning. An alternative is to turn of the flag in org.apache.arrow.memory.BoundsChecking. 

I think the problem of which way is better is a fundamental decision. Maybe the two ways are suitable for different scenarios:

1. checking at the beginning provides both performance and security benefits.
2. disabling the flag provides better code readability (Vector APIs are better to understand and follow than PlatformDependent APIs)

What do you think?

Reply:
I prefer 1 since we have the boundary checks enabled by default (both in arrow and dremio).

For non-trivial code, we can create wrappers over the PlatformDependent code if it helps with readability.

Reply:
@pravindra Changed to 1. Thanks for your suggestion.